crates/wasm3-runtime/src/lib.rs

```diff
@@ -0,0 +1,111 @@
+use anyhow::{anyhow, Error};
+use hotg_rune_runtime::Image;
+use wasm3::{
+    CallContext, Environment, Function, Module, ParsedModule, WasmArgs,
+    WasmType,
+};
+
+const STACK_SIZE: u32 = 1024 * 16;
+
+// FIXME: `wasm3`'s error type isn't Send or Sync since it contains a raw
+// pointer, so it's not `anyhow`-compatible. We work around that by formatting
+// it.
+trait Wasm3ResultExt<T> {
+    fn to_anyhow(self) -> Result<T, anyhow::Error>;
+}
+
+impl<T> Wasm3ResultExt<T> for Result<T, wasm3::error::Error> {
+    fn to_anyhow(self) -> Result<T, anyhow::Error> {
+        self.map_err(|e| anyhow!("{}", e))
+    }
+}
+
+#[derive(Debug)]
+pub struct Runtime {
+    rt: wasm3::Runtime,
+}
+
+impl Runtime {
+    pub fn load<I>(wasm: &[u8], image: I) -> Result<Self, Error>
+    where
+        I: for<'a> Image<Registrar<'a>>,
+    {
+        let env = Environment::new().to_anyhow()?;
+        // XXX note that `ParsedModule::parse` has a soundness bug! `wasm` needs
```

Comment:
Can you copy `wasm` into the `Runtime` struct as a `Vec<u8>` and give `ParsedModule::parse()` a reference to it? That way we work around the soundness bug while also making sure `Runtime: 'static`.

We may need the `ouroboros` or `rental` crates here because we're declaring a self-referential struct and also need to make sure the `rt` field is destroyed before the new `wasm` field.

Other than that it should be good to merge.